src/protocol/client.c

```diff
@@ -80,13 +80,16 @@ protocolClientNew(const String *name, const String *service, IoRead *read, IoWri
     MEM_CONTEXT_NEW_BEGIN("ProtocolClient")
     {
         this = memNew(sizeof(ProtocolClient));
-        this->memContext = memContextCurrent();
 
-        this->name = strDup(name);
-        this->errorPrefix = strNewFmt("raised from %s", strPtr(this->name));
-        this->read = read;
-        this->write = write;
-        this->keepAliveTime = timeMSec();
+        *this = (ProtocolClient)
+        {
+            .memContext = memContextCurrent(),
```

Comment:
In protocolClientNew, shouldn't we be using MEM_CONTEXT_NEW() instead of memContextCurrent()?